qcodes/instrument_drivers/tektronix/TPS2012_channels.py

```diff
@@ -0,0 +1,344 @@
+import logging
+import binascii
+
+import numpy as np
+from pyvisa.errors import VisaIOError
+from functools import partial
+
+from qcodes import VisaInstrument, validators as vals
+from qcodes import InstrumentChannel, ChannelList
+from .TPS2012 import TraceNotReady, ScopeArray
+from qcodes import ArrayParameter
+
+log = logging.getLogger(__name__)
+
+class ScopeArray(ArrayParameter):
+    def __init__(self, name, instrument, channel):
+        super().__init__('scope_measurement', shape=(2500,),
+                         label='Voltage',
+                         unit='V ',
+                         setpoint_names=('Time', ),
+                         setpoint_labels=('Time', ),
+                         setpoint_units=('s',),
+                         docstring='holds an array from scope')
+        self.channel = channel
+        self._instrument = instrument
+
+    def calc_set_points(self):
+        message = self._instrument.ask('WFMPre?')
```

Comment:
There are some conditions under which this call does not return the full expected response. We should ensure that these conditions are not met. (Sorry, I don't remember the details, but it's something like an enabling of the channels)